### DISCUSSION TITLE
<opentracks:distance>

### DISCUSSION THREAD

**Discussion by brianjmurrell** [OpenTracksApp/OpenTracks#1160] [Q&A]
I'm wondering why the use of `<opentracks:distance>` in a GPX file.

Is there no standard GPX track point extension anywhere that describes the distance of a track point?

**Comment by dennisguse** [comment]
To the best of my knowledge no. At least default GPX and the Garmin provided schemas don't. 

For GPX (also KML),  the distance is usually derived from the GPS coordinates. However, if you use a specific sensor (e.g. for cycling), which has a higher precision than GPS, then this sensor data needs to be exported as well - at least to be able to import it again and restore the statistics.

For KML, we even support exporting tracks that do not contain GPS data (e.g. indoor training or GPS turned off). Sadly, this is not allowed by GPX (coordinates are mandatory).

That's why there is `opentracks:distance`.

**Reply by brianjmurrell** [reply]
I just came across http://www.cluetrust.com/Schemas/gpxdata10.xsd.  Does that sufficiently provide a [defacto] "standard" for distance that might be more widely understood by GPX importing software?

> For GPX (also KML), the distance is usually derived from the GPS coordinates.

Of course, that only works when you are actually covering physical distance.  When you are stationary, with a sensor that reports distance the distance tags are needed.

> For KML, we even support exporting tracks that do not contain GPS data

What tag is that format using for distance?  Still `opentracks:distance`?

FWIW, it seems that the link above also adds `dataPt` for being able to add data points without GPS data.

**Comment by brianjmurrell** [comment]
Additionally, I just asked cluetrust.com and they said that that distance value is whatever value the GPS provides at the reading time, which ends up being cumulative for most GPS devices.

Is there any chance of adding this http://www.cluetrust.com/Schemas/gpxdata10.xsd `distance` tag to GPX exports?

**Reply by dennisguse** [reply]
In OpenTracks case, we do not accumulate th distance from GPS to be stored in this tag. It only contains the distance that was provided by a sensor.
Don't know what GPS devices might provide here as we only access the high-level Android API.

**Comment by brianjmurrell** [comment]
So when you do read distance from the high-level Android API, is each reading cumulative relative the previous reading?

**Reply by dennisguse** [reply]
We just get lat and lng per location and then use a Android provided function to compute the distance.

**Comment by brianjmurrell** [comment]
#1169
